test/io/test_io.cpp

```diff
@@ -661,6 +661,189 @@ TEST (PCL, IO)
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST (PCL, EmptyCloudToPCD)
+{
+  pcl::PointCloud<pcl::PointXYZ> cloud;
+  int res = pcl::io::savePCDFileASCII("ascii.pcd", cloud);
+  EXPECT_EQ (res, 0);
+  pcl::PointCloud<pcl::PointXYZ> cloud_in_ascii;
+  cloud_in_ascii.width = 10; // Make sure loadPCDFile overwrites this
+  cloud_in_ascii.height = 10; // Make sure loadPCDFile overwrites this
+  pcl::io::loadPCDFile("ascii.pcd", cloud_in_ascii);
+  remove("ascii.pcd");
+  EXPECT_EQ(cloud.width, cloud_in_ascii.width);
+  EXPECT_EQ(cloud.height, cloud_in_ascii.height);
+  
+  pcl::Indices indices;
+  res = pcl::io::savePCDFile("ascii_indices.pcd", cloud, indices);
+  EXPECT_EQ (0, res);
+  pcl::PointCloud<pcl::PointXYZ> cloud_in_indices;
+  cloud_in_indices.width = 10; // Make sure loadPCDFile overwrites this
+  pcl::io::loadPCDFile("ascii_indices.pcd", cloud_in_indices);
+  remove("ascii_indices.pcd");
+  EXPECT_EQ(cloud.width, cloud_in_indices.width);
+  EXPECT_EQ(1, cloud_in_indices.height); // if we specify indices height must be 1
+  
+  res = pcl::io::savePCDFileBinary("binary.pcd", cloud);
+  EXPECT_EQ (0, res);
+  pcl::PointCloud<pcl::PointXYZ> cloud_in_binary;
+  cloud_in_binary.width = 10; // Make sure loadPCDFile overwrites this
+  cloud_in_binary.height = 10; // Make sure loadPCDFile overwrites this
+  pcl::io::loadPCDFile("binary.pcd", cloud_in_binary);
+  remove("binary.pcd");
+  EXPECT_EQ(cloud.width, cloud_in_binary.width);
+  EXPECT_EQ(cloud.height, cloud_in_binary.height);
+
+  res = pcl::io::savePCDFileBinaryCompressed("binary_compressed.pcd", cloud);
+  EXPECT_EQ (0, res);
+  pcl::PointCloud<pcl::PointXYZ> cloud_in_compressed;
+  cloud_in_compressed.width = 10; // Make sure loadPCDFile overwrites this
+  cloud_in_compressed.height = 10; // Make sure loadPCDFile overwrites this
+  pcl::io::loadPCDFile("binary_compressed.pcd", cloud_in_compressed);
+  remove("binary_compressed.pcd");
+  EXPECT_EQ(cloud.width, cloud_in_compressed.width);
+  EXPECT_EQ(cloud.height, cloud_in_compressed.height);
+
+  // Data initialization for pcl::PCLPointCloud2 interface
+  pcl::PCLPointCloud2 cloud2;
+  pcl::PCLPointField x, y, z;
+  x.name = "x";
+  x.datatype = pcl::PCLPointField::FLOAT32;
+  y.name = "y";
+  y.datatype = pcl::PCLPointField::FLOAT32;
+  z.name = "z";
+  z.datatype = pcl::PCLPointField::FLOAT32;
+  cloud2.fields.push_back(x);
+  cloud2.fields.push_back(y);
+  cloud2.fields.push_back(z);
+  cloud2.is_dense = true;
+
+  res = pcl::io::savePCDFile ("ascii_pc2.pcd", cloud2,
+                              Eigen::Vector4f::Zero (),
+                              Eigen::Quaternionf::Identity ());
+  EXPECT_EQ (0, res);
+  pcl::PCLPointCloud2 cloud2_in_ascii;
+  cloud2_in_ascii.width = 10;
+  cloud2_in_ascii.height = 10;
+  res = loadPCDFile ("ascii_pc2.pcd", cloud2_in_ascii);
+  remove("ascii_pc2.pcd");
+  EXPECT_NE (-1, res);
```

Comment:
Why are you sometimes testing for `res == 0` and sometimes for `res != -1`? In both cases you want to make sure that the function was successful, right? I would prefer testing for equality to zero

Reply:
And could you also please test the return value of all `loadPCDFile` calls?

Reply:
@mvieth Thank you for your kindly review.
> Why are you sometimes testing for `res == 0` and sometimes for `res != -1`? In both cases you want to make sure that the function was successful, right? I would prefer testing for equality to zero

I obey existing test behaver like TEST (PCL, IO).

As a possible reason why the existing test is checking at -1, I imagined that the read function would really want to return the read size on success, not 0.
I don't know what the intention was, but I decided that it would be better to follow the existing test case and check for -1.

In light of the above, would 0 be better?
I do not know the design intent, so I will follow your directions.

> And could you also please test the return value of all `loadPCDFile` calls?

Sure.

Reply:
[The documentation](https://github.com/PointCloudLibrary/pcl/blob/master/io/include/pcl/io/pcd_io.h#L274) says that the read functions always return 0 on success. So I would say it is best to test for `res == 0` if you expect that the read/load functions succeeded. This is a "stronger" test than `res != -1`.

Reply:
Thank you for your comment, I fixed it to 0.